core/src/sumeragi/main_loop.rs

```diff
@@ -842,17 +875,19 @@ fn reset_state(
         // a view change a new block is immediately created by the leader
         *round_start_time = Instant::now();
         was_commit_or_view_change = true;
+        *old_view_change_index = 0;
     }
 
-    if current_view_change_index != *old_view_change_index {
-        current_topology.rebuild_with_new_view_change_count(current_view_change_index);
+    while *old_view_change_index < current_view_change_index {
+        *old_view_change_index += 1;
+        error!(addr=%peer_id.address, "Rotating the entire topology.");
+        current_topology.rotate_all();
```

Comment:
what is the advantage of rotating entire topology `current_view_change_index - old_view_change_index` times vs only once?

Reply:
is it so that all peers reach the same topology? and if you didn't do it many times they could reach different topologies?

Reply:
It is so that all peers reach the same topology. And if we don't do it the same amount of times on each peer then they reach different topologies and cannot reach consensus. You are correct.